src/utils/orgUtils.ts

```diff
@@ -65,6 +72,103 @@ export class OrgUtils {
         }
     }
 
+    public static async getAllCompactLayoutsForSObject(
+        sObjectName: string
+    ): Promise<any> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts`
+        );
+
+        return Promise.resolve(result);
+    }
+
+    public static async getCompactLayoutForSObject(
+        sObjectName: string,
+        recordTypeId: string
+    ): Promise<any> {
+        const org = await Org.create();
+        const conn = org.getConnection();
+
+        const result = await conn.request(
+            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts/${recordTypeId}`
+        );
+
+        return Promise.resolve(result);
+    }
```

Comment:
I'm wrapping `conn.request` because `request` method of `Connection` returns `StreamResult`. `StreamResult` is a type from `@salesforce/core` API and I couldn't find a way to create it or use it. Worse, Typescript was very strict when I tried to create stubs using sinon. So I wrapped them in methods and this way I can wrap the wrapping methods in the unit tests.

Reply:
It looks like it returns a `StreamPromise` from some version of the `jsforce` package—which is just an extension of `Promise` that has some stream-specific functions added.

But I think it's fine you're doing it this way, in any case. As I mentioned below, I would change the type to `Promise<Object>`, and coerce that type as necessary.

If we wanted to get fancy at some point, the `request` method takes a template type, so we could be more definitive in specifying what we want to return, for example:

```typescript
const result: Promise<OurStronglyTypedCompactLayoutResultType> =
    await conn.request<OurStronglyTypedCompactLayoutResultType>(
            `/services/data/v59.0/sobjects/${sObjectName}/describe/compactLayouts/${recordTypeId}`
    );
return Promise.resolve(result);
```